hyper-lib/src/node.rs

```diff
@@ -808,36 +841,37 @@ impl Node {
 
                     // Compute the local difference and remote difference between the sets (what we are missing and they are missing respectively)
                     let peer_recon_state = peer.get_tx_reconciliation_state().unwrap();
-                    let (mut local_diff, remote_diff) =
-                        peer_recon_state.compute_sketch_diff(sketch);
-                    let in_delayed_set = peer_recon_state.get_delayed_set();
-
-                    // Flag the peer as knowing the transaction if we both know it (remote_diff is 1 XOR 1 AND 1.
-                    // Any other of the two partial knowing cases (local_diff or remote_diff set, but not both) will be flagged either
-                    // when we send them the transactions or when they send them to us
-                    if !remote_diff && peer_recon_state.get_recon_set() {
-                        self.get_peer_mut(&peer_id).unwrap().add_known_transaction();
+                    let (mut request_tx, offer_tx) = peer_recon_state.compute_sketch_diff(sketch);
+
+                    // There are two cases in where, even if we know the transaction, we should be requesting it to prevent probing:
+                    // (request_tx = in_local_set XOR in_remote_set AND in_remote_set = 1 XOR 0 AND 1).
+                    // - If we have chosen this peer for fanout, but we have still not announced it (timer hasn't ticked yet)
+                    // - If the transaction is still in their delayed set
+                    if request_tx && self.knows_transaction() && peer.we_announced_tx() {
```

Comment:
the second condition is redundant here? just checking.

Reply:
could be asserted instead I guess :)

Reply:
It is indeed. I'll move it to an assertion